app/services/forest_liana/ability/permission/smart_action_approval.rb

```diff
@@ -0,0 +1,67 @@
+require 'jwt'
+
+module ForestLiana
+  module Ability
+    module Permission
+    class SmartActionApproval
+
+      def initialize(parameters, collection, smart_action, user)
+        @parameters = parameters
+        @collection = collection
+        @smart_action = smart_action
+        @user = user
+      end
+
+      def can_trigger?
+        if @smart_action['triggerEnabled'].include?(@user['roleId']) && @smart_action['approvalRequired'].exclude?(@user['roleId'])
+          return true if @smart_action['triggerConditions'].empty? || match_conditions('triggerConditions')
+
+        elsif @parameters[:data][:attributes][:signed_approval_request].present? && @smart_action['userApprovalEnabled'].include?(@user['roleId'])
```

Comment:
I must go buy some 👓

Reply:
But how does it replace the @parameters[:data][:attributes] by the one inside the `signed_approval_request`?

Reply:
We check if the `@parameters[:data][:attributes][:signed_approval_request]` exist and at line 20 we call the `deserialize` method to replace de parameters.